packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1415,9 +1429,8 @@ export class SnapController extends BaseController<
   }
 }
 
-function isValidSnapVersion(version: unknown): version is string {
+function isValidSnapVersionRange(version: unknown): version is string {
   return Boolean(
-    typeof version === 'string' &&
-      (version === 'latest' || isValidSemver(version)),
+    typeof version === 'string' && semver.validRange(version) !== null,
```

Comment:
Is it intentional to remove the option to use `version === 'latest'`

Reply:
Yeah, latest was removed from all places where semver is expected. This is because it's behaviour is not clearly defined.
For example, what should happen when a dapp requests latest while a user has one already installed? In case of npm, that would mean fetching config file and checking it again. In case of local development - well, there's no log of manifests, so what would "latest" mean in that scenario?

Additionally it would break all existing snaps and examples on every update in npm

Reply:
👍